rules/type-declaration/tests/Rector/FunctionLike/ReturnTypeDeclarationRector/Fixture/respect_children_return_type.php.inc

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Rector\TypeDeclaration\Tests\Rector\FunctionLike\ReturnTypeDeclarationRector\Fixture;
+
+class RespectChildrenReturnType
+{
+    public function run()
+    {
+        throw new \Exception("not implemented");
+    }
+}
+
+class RespectChildrenReturnType2 extends RespectChildrenReturnType {    
+    public function run() : Baba {
+       return new Baba();        
+    }
+}
+
+class Baba {
+}
+?>
```

Comment:
Same here, here only the tested class should be present.
The rest should be moved to `/Source` directory.


If you look at other tests, the structure is following:

```bash
SomeTest.php // the test
/Fixture // directory with tested files, that should or should not be refactored
/Source // all other files, that might be needed, simulates /vendor in normal project
/config // if config is needed
```

Reply:
This one is fine just like this I think. I have BaseAuthenticator throwing an exception and children which returns some class. Rector makes the parent return void and then changes the return type of the other one to void too.